### ISSUE TITLE
`k_msgq_put` does not wake up higher priority thread waiting on message queue with `k_poll`

### ISSUE THREAD

**Issue by jkivilin** [zephyrproject-rtos/zephyr#79617]
**Describe the bug**

My application is using message queues with CONFIG_POLL. Project uses cooperative scheduling only and is build for embedded ARM target (Cortex-M). I noticed that placing new event on message queue with `k_msgq_put` does not immediately wake up higher priority thread polling on same message queue. 

I think issue that `k_msgq_put` is missing rescheduling on CONFIG_POLL path. When comparing `k_msgq_put` and `k_sem_give`, I see that message queue CONFIG_POLL code does not request rescheduling the same way is semaphore code path does:
 - message queue: https://github.com/zephyrproject-rtos/zephyr/blob/029540abece01361f52025534594a0b333cf0afd/kernel/msg_q.c#L161
 - semaphore: https://github.com/zephyrproject-rtos/zephyr/blob/029540abece01361f52025534594a0b333cf0afd/kernel/sem.c#L111

As workaround for my application, I've used dummy `k_sem_give` call after `k_msgq_put` call to enforce rescheduling.

**To Reproduce**

I used following test to reproduce issue outside my application code. 

```
K_MSGQ_DEFINE(msgq_high_prio_thread, sizeof(unsigned int), 4, 4);

static K_THREAD_STACK_DEFINE(high_prio_stack_area, 4096);
static struct k_thread high_prio_data;

static void high_prio_main(void *param1, void *param2, void *param3)
{
    static struct k_poll_event poll_events[1];

    // Setup wake-up for message queue
    k_poll_event_init(&poll_events[0],
                      K_POLL_TYPE_MSGQ_DATA_AVAILABLE,
                      K_POLL_MODE_NOTIFY_ONLY,
                      &msgq_high_prio_thread);

    // Wait for data on message queue
    // Note: buggy, does not wake up by k_msgq_put() even though low priority thread put message to queue.
    (void)k_poll(poll_events, 1, K_FOREVER);

    if (poll_events[0].state == K_POLL_STATE_MSGQ_DATA_AVAILABLE)
    {
        printf("success!");
    }
}

int main(void)
{
    int low_prio_thread_priority = 1;
    int high_prio_thread_priority = -2; // Coop priority
    unsigned int data_to_high_prio = 0x1234;

    k_thread_priority_set(k_current_get(), low_prio_thread_priority);

    // Create high priority thread
    (void)k_thread_create(&high_prio_data, high_prio_stack_area,
                          K_THREAD_STACK_SIZEOF(high_prio_stack_area),
                          high_prio_main,
                          NULL, NULL, NULL,
                          high_prio_thread_priority, 0, K_NO_WAIT);
    k_sleep(K_MSEC(1));

    // Send message to high-priority thread
    (void)k_msgq_put(&msgq_high_prio_thread, &data_to_high_prio, K_NO_WAIT);

    // Uncomment following line for enable rescheduling workaround
    //{ struct k_sem dummy_sem; k_sem_init(&dummy_sem, 1, 1); k_sem_give(&dummy_sem); }

    // Busy loop to prevent rescheduling to high-priority task (should have rescheduled inside k_msgq_put())
    while (1);
}
```

**Expected behavior**

I expect `k_msgq_put()` to reschedule to higher priority thread that is waiting on target message queue using `k_poll`.

**Impact**

As I have workaround, this is just an annoyance.

**Logs and console output**

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: gnuarmemb
 - Commit SHA 029540abece01361f52025534594a0b333cf0afd

**Additional context**
